File: JavaApp/LayoutPractice01/app/src/main/java/com/example/layoutpractice/MainActivity.java
package com.example.layoutpractice;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

// 메인 화면
public class MainActivity extends AppCompatActivity {

    // 뷰가 생성되었을때
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // 그릴 xml 뷰 파일을 연결 시켜준다. (설정한다.)
        setContentView(R.layout.activity_main);
    }
}

File: Other/GraphQL/GraphQLExample/app/src/main/java/com/example/graphqlexample/MainActivity.kt
package com.example.graphqlexample

import FeedResultQuery
import android.content.ContentValues.TAG
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.apollographql.apollo.ApolloCall
import com.apollographql.apollo.ApolloClient
import com.apollographql.apollo.api.Response
import com.apollographql.apollo.exception.ApolloException

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val apolloClient = ApolloClient.builder()
            .serverUrl("https://rickandmortyapi.com/graphql/")
            .build()

        apolloClient.query(FeedResultQuery())
            .enqueue(object : ApolloCall.Callback<FeedResultQuery.Data>() {
                override fun onResponse(response: Response<FeedResultQuery.Data>) {
                    Log.d(TAG, "onResponse: ${response.data?.characters?.results}")
                }

                override fun onFailure(e: ApolloException) {
                    Log.d(TAG, "onFailure: ")
                }
            })
    }
}
